### PR TITLE
refactor: prevent dist table from invoking `scan`

### PR DESCRIPTION
**Pull request by zhongzc**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Part of https://github.com/GreptimeTeam/greptimedb/issues/2065

At present, `scan_to_stream` will be called on the execution path of Dist Table, and merge scan, not partition exec, is the appropriate physical plan.
Furthermore, calls to Table's `scan` should be avoided, the ultimate goal is to remove this method.

Manually tested.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Review by waynexia — APPROVED** [review]
well done

**Review by fengjiachun — APPROVED** [review]
LGTM

File: src/query/src/dist_plan/planner.rs [hunk 1]
```diff
@@ -72,66 +71,62 @@ impl ExtensionPlanner for DistExtensionPlanner {
...
+        };
+
+        let Ok(peers) = self.get_peers(&table_name).await else {
+            // no peers found, going to execute them locally
+            return fallback(&optimized_input).await;
+        };
+
+        // TODO(ruihang): generate different execution plans for different variant merge operation
+        let schema = optimized_input.schema().as_ref().into();
+        let amended_plan = Self::plan_with_full_table_name(input_plan.clone(), &table_name)?;
```
Comment (zhongzc) [inline src/query/src/dist_plan/planner.rs:104]: cc @waynexia, pass `optimized_input` or `input_plan`?

Comment (waynexia) [reply src/query/src/dist_plan/planner.rs:105]: It's intended. The purpose is to leave the original plan to datanode to optimize

**Pull request merged**